lib/routing/sfRoute.class.php

```diff
@@ -858,4 +858,35 @@ public function unserialize($data)
     list($this->tokens, $this->defaultOptions, $this->options, $this->pattern, $this->staticPrefix, $this->regex, $this->variables, $this->defaults, $this->requirements, $this->suffix, $this->customToken) = unserialize($data);
     $this->compiled = true;
   }
+
+    public function jsonSerialize(): array
+    {
+        return [
+            'tokens'         => $this->tokens,
+            'defaultOptions' => $this->defaultOptions,
+            'options'        => $this->options,
+            'pattern'        => $this->pattern,
+            'staticPrefix'   => $this->staticPrefix,
+            'regex'          => $this->regex,
+            'variables'      => $this->variables,
+            'defaults'       => $this->defaults,
+            'requirements'   => $this->requirements,
+            'suffix'         => $this->suffix,
+            'customToken'    => $this->customToken,
+        ];
+    }
+
+    public static function jsonUnserialize(array $raw): self
+    {
+        $rebuilt = new self(
+            $raw['pattern'],
+            $raw['defaults'],
+            $raw['requirements'],
+            $raw['options'],
+        );
+
+        $rebuilt->compile();
+
+        return $rebuilt;
+    }
```

Comment:
I'm confused on this. Why not replicate the same as `unserialize()` where we assign all properties?

Reply:
Because the remaining properties are all `protected`, and cannot easily be accessed without introducing a bunch of getters and setters. I'm trying to avoid that as much as possible, so that we don't have more conflicts to resolve when pulling in changes from the upstream. If you think it worth adding these extra properties then we'll have to solve that for sure - so far I've not noticed anything going wrong with this solution.

Reply:
I will test and see investigate whether these properties holds any meaningful information